tools/dbslower.py

```diff
@@ -101,17 +174,61 @@
 }
 """
 
-usdts = map(lambda pid: USDT(pid=pid), args.pids)
-for usdt in usdts:
-    usdt.enable_probe("query__start", "probe_start")
-    usdt.enable_probe("query__done", "probe_end")
+if args.path and not args.pids:
+    # Uprobes mode
+    program = program_uprobe
+
+    symbols = subprocess.check_output(["nm", "-aD", args.path])
```

Comment:
I think you want [`BPF.get_user_functions_and_addresses`](https://github.com/iovisor/bcc/blob/master/src/python/bcc/__init__.py#L770). It's not Pythonian, it's a wrapper on top of libbcc, which uses libelf for this.